app/queries/drug_stocks_query.rb

```diff
@@ -0,0 +1,119 @@
+class DrugStocksQuery
+  def initialize(facilities:, for_end_of_month:)
+    @facilities = facilities
+    @for_end_of_month = for_end_of_month
+    # assuming that all facilities on the page have the same protocol
+    @protocol = @facilities.first.protocol
+  end
+
+  def protocol_drugs_by_category
+    @drug_categories ||= @protocol.protocol_drugs
```

Comment:
Let's name the caching var the same as the method for consistency.

Reply:
Ah, this must've slipped while refactoring. Will fix, thanks!